project_task_stage_change_restriction/models/project_task_type.py

```
@@ -0,0 +1,37 @@
+
+    allow_assigned_only = fields.Boolean(string="Assigned Only")
+    allow_project_manager = fields.Boolean(string="Project Manager")
+    allow_group_ids = fields.Many2many(
+        "res.groups",
+        "project_task_stage_group_rel",
+        "stage_id",
+        "group_id",
+        string="Group Members",
+    )
+
+    def has_restrictions(self):
```

Ruff (0.11.9)

Missing return type annotation for public function `has_restrictions`

(ANN201)